[WGSL] Convert more wgslc tests to API tests

tadeuzagallo opened the pull request:
#### 73ec86b9cb84767c58f1ff7d36ad1ed3a3e50383
<pre>
[WGSL] Convert more wgslc tests to API tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=295351">https://bugs.webkit.org/show_bug.cgi?id=295351</a>
<a href="https://rdar.apple.com/154885034">rdar://154885034</a>

Reviewed by Mike Wyrzykowski.

Continue the process of moving wgslc tests to TestWebKitAPI so they run
automatically in EWS.

* Source/WebGPU/WGSL/GlobalVariableRewriter.cpp:
(WGSL::RewriteGlobalVariables::run):
(WGSL::RewriteGlobalVariables::argumentBufferStructName):
* Source/WebGPU/WGSL/TypeCheck.cpp:
(WGSL::TypeChecker::declareBuiltins):
* Source/WebGPU/WGSL/tests/access-expression-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/asan-disallow-user-poisoning.wgsl: Removed.
* Source/WebGPU/WGSL/tests/atomics-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/attribute-validation-during-type-checking.wgsl: Removed.
* Source/WebGPU/WGSL/tests/bitcast.wgsl: Removed.
* Source/WebGPU/WGSL/tests/break.wgsl: Removed.
* Source/WebGPU/WGSL/tests/compound-assignment.wgsl: Removed.
* Source/WebGPU/WGSL/tests/continue.wgsl: Removed.
* Source/WebGPU/WGSL/tests/empty-struct.wgsl: Removed.
* Source/WebGPU/WGSL/tests/empty.wgsl: Removed.
* Source/WebGPU/WGSL/tests/f16-literal.wgsl: Removed.
* Source/WebGPU/WGSL/tests/f16-types.wgsl: Removed.
* Source/WebGPU/WGSL/tests/for-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/global-constant-vector.wgsl: Removed.
* Source/WebGPU/WGSL/tests/hex-double-lchar.wgsl: Removed.
* Source/WebGPU/WGSL/tests/if-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/limits-function-vars.wgsl: Removed.
* Source/WebGPU/WGSL/tests/limits-private-vars.wgsl: Removed.
* Source/WebGPU/WGSL/tests/limits-workgroup-vars.wgsl: Removed.
* Source/WebGPU/WGSL/tests/lit.cfg: Removed.
* Source/WebGPU/WGSL/tests/local-constant-vector.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pack-unpack.wgsl: Removed.
* Source/WebGPU/WGSL/tests/phony-assignment.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pointer-as-constant.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pointer-rewriting-concrete-type.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pointers-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/redeclaration-reordering.wgsl: Removed.
* Source/WebGPU/WGSL/tests/redeclaration-type-checking.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-self-const.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-self-function.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-self-struct.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-struct-const.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-redeclaration.wgsl: Removed.
* Source/WebGPU/WGSL/tests/requirements.txt: Removed.
* Source/WebGPU/WGSL/tests/return-type-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/shadowing-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/struct-with-duplicate-member.wgsl: Removed.
* Source/WebGPU/WGSL/wgslc.cpp:
(runWGSL):
* Tools/TestWebKitAPI/Tests/WGSL/MetalCompilationTests.mm:
(TestWGSLAPI::testCompilation):
(TestWGSLAPI::expectPrepareError):
(TestWGSLAPI::expectGenerateError):
(TestWGSLAPI::WGSLMetalCompilationTests::SetUpTestSuite):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Empty)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ReturnTypePromotion)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstantFunctionNotInOutput)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, BuiltinAliases)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, AccessExpression)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayAliasConstructor)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayCountExpression)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLengthPointer)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLengthPointer2)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLengthUnorderd)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLength)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayPrimitiveStruct)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayVec3)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Atomics)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Attributes)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, BindingUIntMax)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Concretization)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstAssert)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstantsUtf16)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstantsUtf8)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Division)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, For)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, FragmentOutput)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, FuzzerTests)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, GlobalConstantVector)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, GlobalOrdering)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, GlobalSameBinding)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, HexDoubleLChar)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, HexDoubleUChar)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, If)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, LargeStruct)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Limits)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, LocalConstantVector)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Loop)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, MinusMinusAmbiguity)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Overload)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Override)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PackUnpack)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PackingNestedArray)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PackingPointerArguments)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PointerRewritingConcreteType)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Pointers)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, References)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Reordering)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, RuntimeSizedArray)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Scope)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Shadowing)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Struct)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Switch)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Swizzle)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, TextureExternal)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, TypePromotion)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, While)):
(TestWGSLAPI::checkLiteral): Deleted.
(TestWGSLAPI::checkNot): Deleted.
(TestWGSLAPI::TEST(WGSLMetalCompilationTests, ReturnTypePromotion)): Deleted.
(TestWGSLAPI::TEST(WGSLMetalCompilationTests, ConstantFunctionNotInOutput)): Deleted.
* Tools/TestWebKitAPI/Tests/WGSL/TestWGSLAPI.cpp:
* Tools/TestWebKitAPI/Tests/WGSL/TestWGSLAPI.h:
(TestWGSLAPI::fn):
(TestWGSLAPI::enableF16):
(TestWGSLAPI::file):
(TestWGSLAPI::checkLiteral):
(TestWGSLAPI::checkNot):
(TestWGSLAPI::check):
(TestWGSLAPI::staticCheck):
(TestWGSLAPI::prepare):
(TestWGSLAPI::generate):
(TestWGSLAPI::performChecks):
* Tools/TestWebKitAPI/Tests/WGSL/TypeCheckingTests.cpp:
(TestWGSLAPI::expectNoError):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Array)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, AttributeValidation)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, AccessExpression)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Atomics)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Bitcast)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Break)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, CompoundAssignment)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Continue)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, EmptyStruct)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, F16)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, For)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Fuzz_133788509)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, If)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Limits)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, PhonyAssignment)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, PointerAsConstant)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, PointerAccessMode)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Redeclaration)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Reordering)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Return)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, ParameterScope)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, StructWithDuplicateMember)):
(TestWGSLAPI::fn): Deleted.
(TestWGSLAPI::TEST(WGSLMetalGenerationTests, Array)): Deleted.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/access-expression.wgsl: Renamed from Source/WebGPU/WGSL/tests/access-expression.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/aliases.wgsl: Renamed from Source/WebGPU/WGSL/tests/aliases.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-alias-constructor.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-alias-constructor.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-count-expression.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-count-expression.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length-pointer.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length-pointer.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length-pointer2.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length-pointer2.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length-unordered.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length-unordered.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-primitive-struct.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-primitive-struct.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-vec3.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-vec3.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/atomics.wgsl: Renamed from Source/WebGPU/WGSL/tests/atomics.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/attribute-validation.wgsl: Renamed from Source/WebGPU/WGSL/tests/attribute-validation.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/attributes-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/attributes-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/attributes.wgsl: Renamed from Source/WebGPU/WGSL/tests/attributes.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/binding-uint-max.wgsl: Renamed from Source/WebGPU/WGSL/tests/binding-uint-max.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/concretization.wgsl: Renamed from Source/WebGPU/WGSL/tests/concretization.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/const-assert-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/const-assert-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/const-assert.wgsl: Renamed from Source/WebGPU/WGSL/tests/const-assert.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constant-matrix.wgsl: Renamed from Source/WebGPU/WGSL/tests/constant-matrix.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constants-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/constants-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constants-utf16.wgsl: Renamed from Source/WebGPU/WGSL/tests/constants-utf16.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constants-utf8.wgsl: Renamed from Source/WebGPU/WGSL/tests/constants.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/division-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/division-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/division.wgsl: Renamed from Source/WebGPU/WGSL/tests/division.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/for.wgsl: Renamed from Source/WebGPU/WGSL/tests/for.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fragment-output.wgsl: Renamed from Source/WebGPU/WGSL/tests/fragment-output.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/function-call.wgsl: Renamed from Source/WebGPU/WGSL/tests/function-call.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-127229681.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-127229681.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-128785160.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-128785160.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-130082002.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-130082002.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-130088292.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-130088292.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-130092499.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-130092499.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-133788509.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-133788509.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-136222279.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-136222279.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-constant-vector.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-ordering.wgsl: Renamed from Source/WebGPU/WGSL/tests/global-ordering.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-same-binding.wgsl: Renamed from Source/WebGPU/WGSL/tests/global-same-binding.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-used-by-callee.wgsl: Renamed from Source/WebGPU/WGSL/tests/global-used-by-callee.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/hex-double-lchar.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/hex-double-uchar.wgsl: Renamed from Source/WebGPU/WGSL/tests/hex-double-uchar.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/if.wgsl: Renamed from Source/WebGPU/WGSL/tests/if.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/large-struct.wgsl: Renamed from Source/WebGPU/WGSL/tests/large-struct.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-brace-enclosed.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-brace-enclosed.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-composite-type.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-composite-type.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-const-array.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-const-array.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-function-parameters.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-function-parameters.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-function-vars.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-private-vars.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-struct-members.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-struct-members.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-switch-case.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-switch-case.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-workgroup-vars.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/location-uint-max.wgsl: Renamed from Source/WebGPU/WGSL/tests/location-uint-max.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/loop.wgsl: Renamed from Source/WebGPU/WGSL/tests/loop.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/minus-minus-ambiguity.wgsl: Renamed from Source/WebGPU/WGSL/tests/minus-minus-ambiguity.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/modulo.wgsl: Renamed from Source/WebGPU/WGSL/tests/modulo.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/name-mangling.wgsl: Renamed from Source/WebGPU/WGSL/tests/name-mangling.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/overload-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/overload-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/overload.wgsl: Renamed from Source/WebGPU/WGSL/tests/overload.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/override.wgsl: Renamed from Source/WebGPU/WGSL/tests/override.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/pack-unpack.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/packing-nested-array.wgsl: Renamed from Source/WebGPU/WGSL/tests/packing-nested-array.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/packing-pointer-arguments.wgsl: Renamed from Source/WebGPU/WGSL/tests/packing-pointer-arguments.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/packing.wgsl: Renamed from Source/WebGPU/WGSL/tests/packing.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/parse-pointer-assignment.wgsl: Renamed from Source/WebGPU/WGSL/tests/parse-pointer-assignment.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/pointers.wgsl: Renamed from Source/WebGPU/WGSL/tests/pointers.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/references-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/references-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/references.wgsl: Renamed from Source/WebGPU/WGSL/tests/references.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/reordering.wgsl: Renamed from Source/WebGPU/WGSL/tests/reordering.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/required-alignment.wgsl: Renamed from Source/WebGPU/WGSL/tests/required-alignment.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/runtime-sized-array-resource.wgsl: Renamed from Source/WebGPU/WGSL/tests/runtime-sized-array-resource.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/scope.wgsl: Renamed from Source/WebGPU/WGSL/tests/scope.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/shadowing.wgsl: Renamed from Source/WebGPU/WGSL/tests/shadowing.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/struct-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/struct-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/struct.wgsl: Renamed from Source/WebGPU/WGSL/tests/struct.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/switch-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/switch-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/switch.wgsl: Renamed from Source/WebGPU/WGSL/tests/switch.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/swizzle.wgsl: Renamed from Source/WebGPU/WGSL/tests/swizzle.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/texture-external.wgsl: Renamed from Source/WebGPU/WGSL/tests/texture-external.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/texture-gather.wgsl: Renamed from Source/WebGPU/WGSL/tests/texture-gather.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/texture-offset.wgsl: Renamed from Source/WebGPU/WGSL/tests/texture-offset.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/type-promotion.wgsl: Renamed from Source/WebGPU/WGSL/tests/type-promotion.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/types-vs-values.wgsl: Renamed from Source/WebGPU/WGSL/tests/types-vs-values.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/unicode.wgsl: Renamed from Source/WebGPU/WGSL/tests/unicode.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/unterminated-comment.wgsl: Renamed from Source/WebGPU/WGSL/tests/unterminated-comment.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/var-initialization-with-var.wgsl: Renamed from Source/WebGPU/WGSL/tests/var-initialization-with-var.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/visibility.wgsl: Renamed from Source/WebGPU/WGSL/tests/visibility.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/while-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/while-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/while.wgsl: Renamed from Source/WebGPU/WGSL/tests/while.wgsl.

Canonical link: <a href="https://commits.webkit.org/297149@main">https://commits.webkit.org/297149@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f3d0b6dd26083072e6d4f3bf894cc1d8ffe6a253

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110634 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30293 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20726 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116660 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60901 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112597 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30972 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38882 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84125 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113582 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24733 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99616 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64565 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24096 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17756 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60455 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94120 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17815 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119450 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37674 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27980 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93090 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38048 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95887 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92913 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23688 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37931 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15675 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33651 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37570 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43040 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37232 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40571 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38940 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c694fdf091d589b6aec3ca140e51b889d8d81385. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5cf61fd7bd928031dda2452ada2b6212748afb03. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/54d122527a990e9d0df2a6b73a2696af117a6f9e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3ab4623d55ae8a9ed63c675798a9dd3baef74a79. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f3d0b6dd26083072e6d4f3bf894cc1d8ffe6a253. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47553

tadeuzagallo committed:
[WGSL] Convert more wgslc tests to API tests
https://bugs.webkit.org/show_bug.cgi?id=295351
rdar://154885034

Reviewed by Mike Wyrzykowski.

Continue the process of moving wgslc tests to TestWebKitAPI so they run
automatically in EWS.

* Source/WebGPU/WGSL/GlobalVariableRewriter.cpp:
(WGSL::RewriteGlobalVariables::run):
(WGSL::RewriteGlobalVariables::argumentBufferStructName):
* Source/WebGPU/WGSL/TypeCheck.cpp:
(WGSL::TypeChecker::declareBuiltins):
* Source/WebGPU/WGSL/tests/access-expression-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/asan-disallow-user-poisoning.wgsl: Removed.
* Source/WebGPU/WGSL/tests/atomics-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/attribute-validation-during-type-checking.wgsl: Removed.
* Source/WebGPU/WGSL/tests/bitcast.wgsl: Removed.
* Source/WebGPU/WGSL/tests/break.wgsl: Removed.
* Source/WebGPU/WGSL/tests/compound-assignment.wgsl: Removed.
* Source/WebGPU/WGSL/tests/continue.wgsl: Removed.
* Source/WebGPU/WGSL/tests/empty-struct.wgsl: Removed.
* Source/WebGPU/WGSL/tests/empty.wgsl: Removed.
* Source/WebGPU/WGSL/tests/f16-literal.wgsl: Removed.
* Source/WebGPU/WGSL/tests/f16-types.wgsl: Removed.
* Source/WebGPU/WGSL/tests/for-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/global-constant-vector.wgsl: Removed.
* Source/WebGPU/WGSL/tests/hex-double-lchar.wgsl: Removed.
* Source/WebGPU/WGSL/tests/if-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/limits-function-vars.wgsl: Removed.
* Source/WebGPU/WGSL/tests/limits-private-vars.wgsl: Removed.
* Source/WebGPU/WGSL/tests/limits-workgroup-vars.wgsl: Removed.
* Source/WebGPU/WGSL/tests/lit.cfg: Removed.
* Source/WebGPU/WGSL/tests/local-constant-vector.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pack-unpack.wgsl: Removed.
* Source/WebGPU/WGSL/tests/phony-assignment.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pointer-as-constant.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pointer-rewriting-concrete-type.wgsl: Removed.
* Source/WebGPU/WGSL/tests/pointers-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/redeclaration-reordering.wgsl: Removed.
* Source/WebGPU/WGSL/tests/redeclaration-type-checking.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-self-const.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-self-function.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-self-struct.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle-struct-const.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-cycle.wgsl: Removed.
* Source/WebGPU/WGSL/tests/reordering-redeclaration.wgsl: Removed.
* Source/WebGPU/WGSL/tests/requirements.txt: Removed.
* Source/WebGPU/WGSL/tests/return-type-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/shadowing-errors.wgsl: Removed.
* Source/WebGPU/WGSL/tests/struct-with-duplicate-member.wgsl: Removed.
* Source/WebGPU/WGSL/wgslc.cpp:
(runWGSL):
* Tools/TestWebKitAPI/Tests/WGSL/MetalCompilationTests.mm:
(TestWGSLAPI::testCompilation):
(TestWGSLAPI::expectPrepareError):
(TestWGSLAPI::expectGenerateError):
(TestWGSLAPI::WGSLMetalCompilationTests::SetUpTestSuite):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Empty)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ReturnTypePromotion)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstantFunctionNotInOutput)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, BuiltinAliases)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, AccessExpression)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayAliasConstructor)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayCountExpression)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLengthPointer)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLengthPointer2)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLengthUnorderd)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayLength)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayPrimitiveStruct)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ArrayVec3)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Atomics)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Attributes)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, BindingUIntMax)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Concretization)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstAssert)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstantsUtf16)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, ConstantsUtf8)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Division)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, For)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, FragmentOutput)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, FuzzerTests)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, GlobalConstantVector)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, GlobalOrdering)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, GlobalSameBinding)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, HexDoubleLChar)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, HexDoubleUChar)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, If)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, LargeStruct)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Limits)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, LocalConstantVector)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Loop)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, MinusMinusAmbiguity)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Overload)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Override)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PackUnpack)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PackingNestedArray)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PackingPointerArguments)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, PointerRewritingConcreteType)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Pointers)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, References)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Reordering)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, RuntimeSizedArray)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Scope)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Shadowing)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Struct)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Switch)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, Swizzle)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, TextureExternal)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, TypePromotion)):
(TestWGSLAPI::TEST_F(WGSLMetalCompilationTests, While)):
(TestWGSLAPI::checkLiteral): Deleted.
(TestWGSLAPI::checkNot): Deleted.
(TestWGSLAPI::TEST(WGSLMetalCompilationTests, ReturnTypePromotion)): Deleted.
(TestWGSLAPI::TEST(WGSLMetalCompilationTests, ConstantFunctionNotInOutput)): Deleted.
* Tools/TestWebKitAPI/Tests/WGSL/TestWGSLAPI.cpp:
* Tools/TestWebKitAPI/Tests/WGSL/TestWGSLAPI.h:
(TestWGSLAPI::fn):
(TestWGSLAPI::enableF16):
(TestWGSLAPI::file):
(TestWGSLAPI::checkLiteral):
(TestWGSLAPI::checkNot):
(TestWGSLAPI::check):
(TestWGSLAPI::staticCheck):
(TestWGSLAPI::prepare):
(TestWGSLAPI::generate):
(TestWGSLAPI::performChecks):
* Tools/TestWebKitAPI/Tests/WGSL/TypeCheckingTests.cpp:
(TestWGSLAPI::expectNoError):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Array)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, AttributeValidation)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, AccessExpression)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Atomics)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Bitcast)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Break)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, CompoundAssignment)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Continue)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, EmptyStruct)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, F16)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, For)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Fuzz_133788509)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, If)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Limits)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, PhonyAssignment)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, PointerAsConstant)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, PointerAccessMode)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Redeclaration)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Reordering)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Return)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, ParameterScope)):
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, StructWithDuplicateMember)):
(TestWGSLAPI::fn): Deleted.
(TestWGSLAPI::TEST(WGSLMetalGenerationTests, Array)): Deleted.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/access-expression.wgsl: Renamed from Source/WebGPU/WGSL/tests/access-expression.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/aliases.wgsl: Renamed from Source/WebGPU/WGSL/tests/aliases.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-alias-constructor.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-alias-constructor.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-count-expression.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-count-expression.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length-pointer.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length-pointer.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length-pointer2.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length-pointer2.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length-unordered.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length-unordered.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-length.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-length.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-primitive-struct.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-primitive-struct.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/array-vec3.wgsl: Renamed from Source/WebGPU/WGSL/tests/array-vec3.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/atomics.wgsl: Renamed from Source/WebGPU/WGSL/tests/atomics.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/attribute-validation.wgsl: Renamed from Source/WebGPU/WGSL/tests/attribute-validation.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/attributes-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/attributes-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/attributes.wgsl: Renamed from Source/WebGPU/WGSL/tests/attributes.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/binding-uint-max.wgsl: Renamed from Source/WebGPU/WGSL/tests/binding-uint-max.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/concretization.wgsl: Renamed from Source/WebGPU/WGSL/tests/concretization.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/const-assert-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/const-assert-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/const-assert.wgsl: Renamed from Source/WebGPU/WGSL/tests/const-assert.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constant-matrix.wgsl: Renamed from Source/WebGPU/WGSL/tests/constant-matrix.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constants-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/constants-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constants-utf16.wgsl: Renamed from Source/WebGPU/WGSL/tests/constants-utf16.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/constants-utf8.wgsl: Renamed from Source/WebGPU/WGSL/tests/constants.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/division-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/division-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/division.wgsl: Renamed from Source/WebGPU/WGSL/tests/division.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/for.wgsl: Renamed from Source/WebGPU/WGSL/tests/for.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fragment-output.wgsl: Renamed from Source/WebGPU/WGSL/tests/fragment-output.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/function-call.wgsl: Renamed from Source/WebGPU/WGSL/tests/function-call.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-127229681.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-127229681.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-128785160.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-128785160.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-130082002.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-130082002.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-130088292.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-130088292.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-130092499.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-130092499.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-133788509.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-133788509.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/fuzz-136222279.wgsl: Renamed from Source/WebGPU/WGSL/tests/fuzz-136222279.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-constant-vector.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-ordering.wgsl: Renamed from Source/WebGPU/WGSL/tests/global-ordering.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-same-binding.wgsl: Renamed from Source/WebGPU/WGSL/tests/global-same-binding.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/global-used-by-callee.wgsl: Renamed from Source/WebGPU/WGSL/tests/global-used-by-callee.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/hex-double-lchar.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/hex-double-uchar.wgsl: Renamed from Source/WebGPU/WGSL/tests/hex-double-uchar.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/if.wgsl: Renamed from Source/WebGPU/WGSL/tests/if.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/large-struct.wgsl: Renamed from Source/WebGPU/WGSL/tests/large-struct.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-brace-enclosed.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-brace-enclosed.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-composite-type.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-composite-type.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-const-array.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-const-array.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-function-parameters.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-function-parameters.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-function-vars.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-private-vars.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-struct-members.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-struct-members.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-switch-case.wgsl: Renamed from Source/WebGPU/WGSL/tests/limits-switch-case.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/limits-workgroup-vars.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/location-uint-max.wgsl: Renamed from Source/WebGPU/WGSL/tests/location-uint-max.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/loop.wgsl: Renamed from Source/WebGPU/WGSL/tests/loop.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/minus-minus-ambiguity.wgsl: Renamed from Source/WebGPU/WGSL/tests/minus-minus-ambiguity.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/modulo.wgsl: Renamed from Source/WebGPU/WGSL/tests/modulo.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/name-mangling.wgsl: Renamed from Source/WebGPU/WGSL/tests/name-mangling.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/overload-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/overload-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/overload.wgsl: Renamed from Source/WebGPU/WGSL/tests/overload.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/override.wgsl: Renamed from Source/WebGPU/WGSL/tests/override.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/pack-unpack.wgsl: Added.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/packing-nested-array.wgsl: Renamed from Source/WebGPU/WGSL/tests/packing-nested-array.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/packing-pointer-arguments.wgsl: Renamed from Source/WebGPU/WGSL/tests/packing-pointer-arguments.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/packing.wgsl: Renamed from Source/WebGPU/WGSL/tests/packing.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/parse-pointer-assignment.wgsl: Renamed from Source/WebGPU/WGSL/tests/parse-pointer-assignment.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/pointers.wgsl: Renamed from Source/WebGPU/WGSL/tests/pointers.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/references-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/references-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/references.wgsl: Renamed from Source/WebGPU/WGSL/tests/references.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/reordering.wgsl: Renamed from Source/WebGPU/WGSL/tests/reordering.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/required-alignment.wgsl: Renamed from Source/WebGPU/WGSL/tests/required-alignment.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/runtime-sized-array-resource.wgsl: Renamed from Source/WebGPU/WGSL/tests/runtime-sized-array-resource.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/scope.wgsl: Renamed from Source/WebGPU/WGSL/tests/scope.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/shadowing.wgsl: Renamed from Source/WebGPU/WGSL/tests/shadowing.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/struct-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/struct-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/struct.wgsl: Renamed from Source/WebGPU/WGSL/tests/struct.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/switch-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/switch-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/switch.wgsl: Renamed from Source/WebGPU/WGSL/tests/switch.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/swizzle.wgsl: Renamed from Source/WebGPU/WGSL/tests/swizzle.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/texture-external.wgsl: Renamed from Source/WebGPU/WGSL/tests/texture-external.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/texture-gather.wgsl: Renamed from Source/WebGPU/WGSL/tests/texture-gather.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/texture-offset.wgsl: Renamed from Source/WebGPU/WGSL/tests/texture-offset.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/type-promotion.wgsl: Renamed from Source/WebGPU/WGSL/tests/type-promotion.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/types-vs-values.wgsl: Renamed from Source/WebGPU/WGSL/tests/types-vs-values.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/unicode.wgsl: Renamed from Source/WebGPU/WGSL/tests/unicode.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/unterminated-comment.wgsl: Renamed from Source/WebGPU/WGSL/tests/unterminated-comment.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/var-initialization-with-var.wgsl: Renamed from Source/WebGPU/WGSL/tests/var-initialization-with-var.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/visibility.wgsl: Renamed from Source/WebGPU/WGSL/tests/visibility.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/while-errors.wgsl: Renamed from Source/WebGPU/WGSL/tests/while-errors.wgsl.
* Tools/TestWebKitAPI/Tests/WGSL/shaders/while.wgsl: Renamed from Source/WebGPU/WGSL/tests/while.wgsl.

Canonical link: https://commits.webkit.org/297149@main
